### ISSUE TITLE
Provide x86 pre-built binaries

### ISSUE THREAD

**Issue by Xottab-DUTY** [google/shaderc#1009]

**Comment by zoddicus** [comment]
See https://github.com/google/shaderc/blob/master/downloads.md

**Comment by Xottab-DUTY** [comment]
Before closing this blindly, see downloads.md yourself.

There's no x86 bins, because the kokoro build script is missing x86 configuration completely.

**Comment by Xottab-DUTY** [comment]
No x86 bins for Windows/Linux/MacOS, they all are x64.
Only Android builds scripts have something about x86, but there's no Android builds in the download page, by the way.

**Comment by Xottab-DUTY** [comment]
@zoddicus, reopen this, please. Thank you.

**Comment by zoddicus** [comment]
@dneto0 , the lack of x86 artifacts likely means that the builders are configured as x86-64 builders and are not building any 32-bit code.

**Comment by zoddicus** [comment]
Talked with @dneto0 a bit about this, since the binary download stuff was implemented before I started working on the project.

The binary downloads are just resulting artifacts of the most recent successful builder runs, and have had no significant QA done on them other. They are not really meaningfully supported, and we currently don't have the resources to support them in a more proper way. I am going to write a PR to update the download.md page to more clearly indicate this.

We are not planning on adding/builders 32-bit builders, so will not be adding 32-bit downloads.

**Comment by Xottab-DUTY** [comment]
Why not adding 32-bit builders?
Adding support for 32-bit builders is not just about downloadable untested builds, but also about compilation errors detection and testing.
So, using CI for all supported platforms is normal and it's how it should be, while current situation – is lack of (at least compilation) testing for 32-bit configuration. (at least that's what it looks like from the outside; and if it's true – then lack of testing of testing could turn out badly in future.)

Sorry if I'm being pushy. But, the discussion seems constructive for me and, until it's constructive, it's cool. 😄

**Comment by dneto0** [comment]
The request here is shifting from "providing binaries" to "execute CI".

> Why not adding 32-bit builders?

You didn't mention what OS you're interested in.  Or what compiler.

From what I can tell the LunarG Vulkan SDK's are only offered for 64bit platforms. https://vulkan.lunarg.com/sdk/home 

Similarly, the prebuilt Android NDK is only offered in 64bit form. https://developer.android.com/ndk/downloads 

I would conclude that 32-bit builds are outside the mainstream now, and beyond what we'd like to devote constant attention to.  In my estimation the protective coverage of running 32bit CI on every single commit is below the cost/benefit tradeoff for the risks that I perceive.  This is especially in the context that newer compilers are getting very aggressive about warning against unintended type mismatches on arithmetic and comparisons, and we do turn warnings into errors.

If you find a bug in a 32-bit build, please file an issue and we'll fix it.

Thanks for your interest and engaging with us on this.

**Comment by Xottab-DUTY** [comment]
> From what I can tell the LunarG Vulkan SDK's are only offered for 64bit platforms.

No-no, Vulkan SDK has 32-bit too.
Don't know about Android NDK, though.

> The request here is shifting from "providing binaries" to "execute CI".

Well, currently available pre-built binaries are consequence of executing CI. So, it was implied that providing new binaries requires to execute CI.

> You didn't mention what OS you're interested in. Or what compiler.

Sorry, that's my fault in the first message. My position is to provide builders for all supported configurations where it's possible and it was implied [here](https://github.com/google/shaderc/issues/1009#issuecomment-602656263). Sorry for such implicitness.

> In my estimation the protective coverage of running 32bit CI on every single commit is below the cost/benefit tradeoff for the risks that I perceive.

Ah, that's what the problem!
We can use AppVeyor again, since it's free for open-source projects. (unless there are 'special' cases that I didn't paid attention to)
Why not?
I saw in one of the issues that AppVeyor was used in the past. Why it was refused?

**Comment by dneto0** [comment]
AppVeyor's free tier is very slow, both in latency and runtime.  On the latency front, we were getting turnaround times of about 6 hours, which we very much didn't want to wait for.

We ran into some stability problems on a few projects, especially as we ramped up the number of tests written, especially on debug  builds. If I recall there was a 15 minute limit.  Whatever it was, we were hitting the upper limit.  
Another factor is sometimes we get random failure of the MSVC linker.
